src/contrib/mlir/backend/pass/affine_lowerer.cpp

```diff
@@ -1722,7 +1706,6 @@ namespace
         ArrayRef<Attribute> padAbove = pooling.padAbove().getValue();
 
         Value result = pass.buildOutputDefs(op, rewriter)[0];
-        NGRAPH_CHECK(src && delta && result, "Unexpected null values in MaxPoolBackprop Op");
```

Comment:
Why removing the asserts ?

Reply:
@nmostafa with the change to Value (instead of ValueHandle) the && operator here was invoking a logical and operation in the Affine dialect via overloaded operator&&.  My theory (could be wrong) is that since we are passing mlir::Value around by value it cannot be null which was the check here.  Please correct me if I am wrong.

Reply:
The Value is a  wrapper around a pointer. I do see a bool cast operator. 
https://github.com/llvm/llvm-project/blob/master/mlir/include/mlir/IR/Value.h#L95

You could split this into multiple asserts.

Reply:
I ended up splitting into multiple asserts.